src/main/java/org/opensearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -158,9 +158,49 @@ private class ResolvedIndicesProvider implements IndicesProvider {
         private final ImmutableSet.Builder<String> remoteIndices;
         // set of previously resolved index requests to avoid resolving
         // the same index more than once while processing bulk requests
-        private final Set<MultiKey> alreadyResolved;
+        private final Set<AlreadyResolvedKey> alreadyResolved;
         private final String name;
 
+        private final class AlreadyResolvedKey {
+
+            private final IndicesOptions indicesOptions;
+
+            private final boolean enableCrossClusterResolution;
```

Comment:
Sorry if I am missing something, but how is this field being used? Is this part of the bulk request logic in some way? I am not particularly familiar with this--are we using this field to differentiate between two instances of the same index operation but with different CCR permissions?

Reply:
this is how `enableCrossClusterResolution` this value is set here: https://github.com/opensearch-project/security/blob/1089bd6d60d19088d82f1d6d156652e8f8b804c6/src/main/java/org/opensearch/security/resolver/IndexResolverReplacer.java#L375.
I played with it a bit. If I remove it all tests pass except PIT tests. TBH did not investigate why.

Reply:
No worries.